### PR TITLE
Add functionality for gzipped .jsonl reading and writing

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Add functionality for gzipped .jsonl reading and writing + tests and docs update. Thanks to @RXminuS for submitting a [first PR for this](https://github.com/explosion/srsly/pull/82).
Unrelated formatting changes in readme are due to running `prettier` on it.

### Types of change

New feature.

## Checklist

- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: srsly/tests/test_json_api.py [hunk 1]
```diff
@@ -204,3 +204,44 @@ def test_unsupported_type_error():
...
+    expected = ['{"hello":"world"}\n', '{"test":123}\n', '\n', '{"hello":"world"}\n', '{"test":123}\n']
+    with make_tempdir() as temp_dir:
+        file_path = temp_dir / "tmp.json"
+        write_gzip_jsonl(file_path, data)
+        write_gzip_jsonl(file_path, data, append=True)
+        with gzip.open(file_path, "r") as f:
+            assert [line.decode("utf8") for line in f.readlines()] == expected
+
+def test_read_jsonl_gzip():
+    # todo add test for writing jsonl gzips
```
Comment (adrianeboyd) [inline srsly/tests/test_json_api.py:232]: Is this still a todo?

Comment (rmitsch) [reply srsly/tests/test_json_api.py]: No, copy-past mistake. Exchanged with proper docstring.

File: srsly/tests/cloudpickle/cloudpickle_test.py [hunk 2]
```diff
@@ -872,7 +872,9 @@ def test_builtin_classicmethod(self):
     @pytest.mark.skipif(
         (platform.machine() == "aarch64" and sys.version_info[:2] >= (3, 10))
             or platform.python_implementation() == "PyPy"
-            or (sys.version_info[:2] == (3, 10) and sys.version_info >= (3, 10, 8)),
+            or (sys.version_info[:2] == (3, 10) and sys.version_info >= (3, 10, 8))
+            # Skipping tests on 3.11 due to https://github.com/cloudpipe/cloudpickle/pull/486.
+            or sys.version_info[:2] == (3, 11),
         reason="Fails on aarch64 + python 3.10+ in cibuildwheel, currently unable to replicate failure elsewhere; fails sometimes for pypy on conda-forge; fails for python 3.10.8")
```
Comment (adrianeboyd) [inline srsly/tests/cloudpickle/cloudpickle_test.py:878]:
```suggestion
            # Skipping tests on 3.11 due to https://github.com/cloudpipe/cloudpickle/pull/486.
            or sys.version_info[:2] == (3, 11),
        reason="Fails on aarch64 + python 3.10+ in cibuildwheel, currently unable to replicate failure elsewhere; fails sometimes for pypy on conda-forge; fails for python 3.10.8+ and 3.11")
```

File: srsly/tests/test_json_api.py [hunk 3]
```diff
@@ -204,3 +204,44 @@ def test_unsupported_type_error():
     f = numpy.float32()
     with pytest.raises(TypeError):
         s = json_dumps(f)
+
+
+def test_write_jsonl_gzip():
```
Comment (adrianeboyd) [inline srsly/tests/test_json_api.py:209]: Can you check the formatting on everything?

Comment (rmitsch) [reply srsly/tests/test_json_api.py:216]: Done (didn't apply `black` before, sorry).

**Comment by adrianeboyd** [comment]
Thanks! `append_new_line=True` seems weird, but consistent with the existing methods.

**Pull request merged**